packages/core/src/http-client/http-client.ts

```diff
@@ -187,6 +197,13 @@ function executeWithAxios(request: HttpRequest): Promise<HttpResponse> {
   return axios.request({ ...getAxiosConfigWithDefaults(), ...request });
 }
 
+function executeWithAxiosReturnRequestAndResponse(request: HttpRequest): Promise<HttpRequestAndResponse> {
+  const merged: HttpRequest = { ...getAxiosConfigWithDefaults(), ...request };
```

Comment:
could you use the executeHttpRequest method here.

Reply:
I can extract a function for getting the merged request, but since this is used in the return object, the method call is not avoidable.